empress_gui.py

```diff
@@ -4,362 +4,825 @@
+            self.dup_cost = event.xdata
+            self.trans_cost = event.ydata
+            self.loss_cost = 1.00
+            self.update_compute_reconciliations_btn()
+        else:
+            messagebox.showinfo("Warning", "Please click inside the axes bounds.")
+
+    def dtl_cost(self):
+        """Set DTL costs by clicking on the matplotlib graph or by entering manually."""  
+        self.dup_label = tk.Label(self.costs_frame, text="Duplication:")
+        self.dup_label.grid(row=0, column=0, sticky="w")
         # %P = value of the entry if the edit is allowed
         # see https://stackoverflow.com/questions/4140437/interactively-validating-entry-widget-content-in-tkinter
         dup_vcmd = (self.register(self.validate_dup_input), '%P')
-        self.dup_input = tk.DoubleVar()
-        self.dup_entry_box = tk.Entry(costs_frame, width=3, validate="all", textvariable=self.dup_input, validatecommand=dup_vcmd)
-        
-        dup_label.grid(row=0, column=0, sticky="w")
+        # self.dup_input is tk.DoubleVar(), initialized to be 1.00
+        self.dup_entry_box = CustomEntry(self.costs_frame, width=3, textvariable=self.dup_input)
+        self.dup_entry_box.set_border_color("green")
+        self.dup_entry_box.validate(validate="all", validatecommand=dup_vcmd)
+        #self.dup_entry_box = CustomEntry(self.costs_frame, width=3, validate="all", textvariable=self.dup_input, validatecommand=dup_vcmd)
         self.dup_entry_box.grid(row=0, column=1, sticky="w")
-        self.dup_error.grid(row=0, column=2, sticky="w")
+        self.validate_dup_input(str(1.00))
 
-        trans_label = tk.Label(costs_frame, text="Transfer:")
-        self.trans_error = tk.Label(costs_frame, text="")
+        self.trans_label = tk.Label(self.costs_frame, text="Transfer:")
+        self.trans_label.grid(row=1, column=0, sticky="w")
         # %P = value of the entry if the edit is allowed
         # see https://stackoverflow.com/questions/4140437/interactively-validating-entry-widget-content-in-tkinter
         trans_vcmd = (self.register(self.validate_trans_input), '%P')
-        self.trans_input = tk.DoubleVar()
-        self.trans_entry_box = tk.Entry(costs_frame, width=3, textvariable=self.trans_input, validate="all", validatecommand=trans_vcmd)
-        trans_label.grid(row=1, column=0, sticky="w")
+        # self.trans_input is tk.DoubleVar(), initialized to be 1.00
+        self.trans_entry_box = CustomEntry(self.costs_frame, width=3, textvariable=self.trans_input)
+        self.trans_entry_box.set_border_color("green")
+        self.trans_entry_box.validate(validate="all", validatecommand=trans_vcmd)
+        #self.trans_entry_box = tk.Entry(self.costs_frame, width=3, textvariable=self.trans_input, validate="all", validatecommand=trans_vcmd)
         self.trans_entry_box.grid(row=1, column=1, sticky="w")
-        self.trans_error.grid(row=1, column=2, sticky="w")
+        self.validate_trans_input(str(1.00))
```

Comment:
Changed to self.validate_dup_input(str(self.dup_entry_box.get())). Since the dup_cost is initialized to be 1.00, the validate command has to be triggered for the initialization.

Reply:
Comments added.

Reply:
Sorry I mean dup_input is initialized to be 1.00 in both init and reset function.

Reply:
dup_input is the entry displayed in the entry box. dup_cost is only set after checking that dup_input is valid.